src/Lecture2.hs

```diff
@@ -16,6 +16,8 @@ Unlike exercises to Lecture 1, this module also contains more
 challenging exercises. You don't need to solve them to finish the
 course but you can if you like challenges :)
 -}
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
```

Comment:
Ideally, you shouldn't need to disable this warning. You can export all the data types and top-level functions and the warning will disappear